ipld/resolver/src/service.rs

```diff
@@ -672,46 +674,77 @@ pub fn build_transport(local_key: Keypair) -> Boxed<(PeerId, StreamMuxerBox)> {
 
 async fn download_blob(
     iroh: Iroh,
-    hash: Hash,
+    seq_hash: Hash,
     size: u64,
     node_addr: NodeAddr,
 ) -> anyhow::Result<()> {
+    // Download top-level blob
     // Use an explicit tag so we can keep track of it
-    // TODO: this needs to be tagged with a "user id"
-    let tag = iroh::blobs::Tag(format!("stored-seq-{hash}").into());
-    let res = iroh
-        .blobs()
+    let tag = iroh::blobs::Tag(format!("stored-seq-{seq_hash}").into());
+    iroh.blobs()
         .download_with_opts(
-            hash,
+            seq_hash,
             iroh::client::blobs::DownloadOptions {
                 format: iroh::blobs::BlobFormat::Raw,
```

Comment:
@sanderpick 
this line is the issue, you need to use `iroh::blobs::BlobFormat::HashSeq` to trigger downloading the whole sequence, passing in `Raw` treats this as a single blob